server/hotrod/src/test/scala/org/infinispan/server/hotrod/test/HotRodTestingUtil.scala

```diff
@@ -91,6 +94,19 @@ object HotRodTestingUtil extends Log {
             cfg.transaction().syncCommitPhase(false).syncRollbackPhase(false)
             cfg
          }
+
+         override def getInitializer: ChannelInitializer[Channel] = {
+            // Pass by name since we have circular dependency
+            def getTransport() = {
+               transport
+            }
+            if (configuration.idleTimeout > 0)
+               new HotRodChannelInitializer(this, getTransport(), getEncoder, "test")
+                 with TimeoutEnabledChannelInitializer with SingleByteFrameDecoderChannelInitializer
```

Comment:
It might be better to add a HotRodTestingUtil method that is just like the current one but you can call either based on if this decoder is present or not.